### PR TITLE
Build on PlatformIO

### PR DESCRIPTION
**Pull request by quozl**
Please review; especially;

* [x] is the new README.md correct; it is a merge of two files and has a lot of new stuff,
* [x] do the existing `build/` based methods still build to your satisfaction,
* [x] does the PlatformIO build work for you; it's just `pio run` followed by `pio run -e ENV` where ENV is one of the boards I've tested.

Thanks!

Commit message follows;

- change .gitignore layout to explain what creates some of the files,

- rewrite README.md based on README and add further detail,

- host-serial-linux64 - add a wrapper for using C Forth on a Linux system as a general purpose scripting language,

- add a PlatformIO project configuration file and extra scripts,

- add an Arduino framework platform and app.

Signed-off-by: James Cameron <quozl@laptop.org>

### CODE REVIEW

**Review by MitchBradley — COMMENTED** [review]
I'm getting
```
In file included from src\cforth\dictfile.c:1:0:
src\cforth\forth.h:5:18: fatal error: vars.h: No such file or directory
```
when I do pio run.  I think there might be a problem with the makename step.

**Comment by MitchBradley** [comment]
I got the pio build to work after a lot of drama around ancient versions of platformio and various packages.  Most of my work these days is on windows so my pio installation there is up to date, while my Linux pio installation was old and crufty.  After several rounds of updates, things compiled pretty well.

I did have a final link problem with the featheresp32 build.  Apparently newlib has a "system()" symbol that conflicts with the one in syscalls.c.  I fixed that with this patch:
```
diff --git a/src/cforth/syscall.c b/src/cforth/syscall.c
index 9783a31..8517022 100755
--- a/src/cforth/syscall.c
+++ b/src/cforth/syscall.c
@@ -25,9 +25,11 @@ void prerror(const char *s, cell *up)
     }
 }
 
+#ifndef NOSYSCALL
 cell dosyscall()  { return(unimplemented()); }
 int system() { return(unimplemented()); }
 int chdir()  { return(unimplemented()); }
+#endif
 
 void linemode() {}
 void keymode() {}
```

I ran a few makes in build/* and things behaved pretty much as expected.  There were some glitches but they were probably due to stuff I forgot to check in.

File: README.md [hunk 1]
```diff
@@ -1,12 +1,344 @@
...
+
+There are two build directories that can be used as templates.  You
+can copy these templates to a new name as a starting point for your
+version.
+
+Use `build/host-serial-linux64/Makefile` as an example for how to set
+up a build directory for a host-resident C Forth on a 64-bit Linux
+system.
+
+Use `build/arm-xo-1.75/Makefile` as an example for how to set up a
```
[the remark this answers is not in the document]
Comment (jcameron-sso) [reply README.md:30]: That was inherited from existing `README`, sorry.  Of the embedded targets in `build`, which do you think are a good simple example of things that people could get their hands on?  stm32f103 blue pill?

[the remark this answers is not in the document]
Comment (quozl) [reply README.md:30]: That was inherited from existing README, sorry. Of the embedded targets in
build, which do you think are a good simple example of things that people could
get their hands on? stm32f103 blue pill?

**Comment by quozl** [comment]
Thanks for testing.  The `system()` thing was missed when copying from my private repository.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MitchBradley) [reply README.md:30]: esp8266?

**Comment by MitchBradley** [comment]
Is it ready to merge or do you want to do anything else?

**Comment by jcameron-sso** [comment]
Ready to merge, thanks.  The major public impact for the project is the README.md rewrite, which will change how things look for us on GitHub.  I've looked through the changes again and don't anticipate any regressions.

**Comment by quozl** [comment]
Ready to merge, thanks. The major public impact for the project is the README.md rewrite, which will change how things look for us on GitHub. I've looked through the changes again and don't anticipate any regressions.

**Pull request merged**